src/azure-cli/azure/cli/command_modules/cosmosdb/tests/latest/test_cosmosdb_commands.py

```diff
@@ -1410,10 +1412,80 @@ def test_cosmosdb_managed_service_identity(self, resource_group, key_vault):
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert list(identity_output["userAssignedIdentities"])[0] == id1
+        assert len(identity_output["userAssignedIdentities"]) == 1
+
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg} --identities {id2}').get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert (list(identity_output["userAssignedIdentities"])[0] == id2 or list(identity_output["userAssignedIdentities"])[1] == id2)
+        assert len(identity_output["userAssignedIdentities"]) == 2
+
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg}').get_output_in_json()
+        assert identity_output["type"] == "UserAssigned"
+        assert len(identity_output["userAssignedIdentities"]) == 2
+
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg} --identities {id2}').get_output_in_json()
+        assert identity_output["type"] == "UserAssigned"
+        assert list(identity_output["userAssignedIdentities"])[0] == id1
+        assert len(identity_output["userAssignedIdentities"]) == 1
+
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg} --identities {id2}').get_output_in_json()
+        assert identity_output["type"] == "UserAssigned"
+        assert (list(identity_output["userAssignedIdentities"])[0] == id2 or list(identity_output["userAssignedIdentities"])[1] == id2)
+        assert len(identity_output["userAssignedIdentities"]) == 2
+
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg} --identities {id1} {id2}').get_output_in_json()
+        assert identity_output["type"] == "None"
+
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg} --identities {id1} {id2} [system]').get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert len(identity_output["userAssignedIdentities"]) == 2
+        
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg} --identities {id2}').get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert len(identity_output["userAssignedIdentities"]) == 1
+        
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg} --identities {id1}').get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned"
+        
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg} --identities {id1} {id2} [system]').get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert len(identity_output["userAssignedIdentities"]) == 2
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg} --identities {id1} {id2} [system]').get_output_in_json()
+        assert identity_output["type"] == "None"
+        
```

Comment:
Can you add test cases for the scenario we discussed in teams:
"I found out I'm unable to delete the UA or UAs forever, no matter it's deleting UA or deleting 2 UAs.

Basically the customer will forever unable to reach the "type=None" state. Can I have your help to resolve this? Thank you!"